feast_component/converters.py

```diff
@@ -0,0 +1,57 @@
+import abc
+from typing import Any, Dict, Optional
+
+from google.cloud import bigquery
+from tfx.extensions.google_cloud_big_query import utils
+
+
+class _Converter(abc.ABC):
+    """Takes in data as a dictionary of string -> value and returns the serialized form
+    of whatever representation we want.
+    """
+
+    @abc.abstractmethod
+    def RowToExampleBytes(self, instance: Any) -> bytes:
+        """Generate tf.Example bytes from dictionary.
+
+        Args:
+            instance (Any): Data row generated from data source
+
+        Returns:
+            bytes: Serialized tf.SequenceExample
+        """
+        pass
+
+    @abc.abstractmethod
+    def RowToSequenceExampleBytes(self, instance: Any) -> bytes:
+        """Generate tf.SequenceExample bytes from dictionary.
+
+        Args:
+            instance (Any): Data row generated from data source
+
+        Returns:
+            bytes: Serialized tf.SequenceExample
+        """
+        pass
+
+
+class _BQConverter(_Converter):
+    """Converter class for BigQuery source data"""
+
+    def __init__(self, query: str, project: Optional[str]) -> None:
+        client = bigquery.Client(project=project)
+        # Dummy query to get the type information for each field.
+        query_job = client.query("SELECT * FROM ({}) LIMIT 0".format(query))
```

Comment:
[`LIMIT 0` returns 0 rows](https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#limit_and_offset_clause) - why would we want that?

Reply:
Mostly based on https://github.com/tensorflow/tfx/blob/master/tfx/extensions/google_cloud_big_query/example_gen/executor.py#L41